src/main/java/org/embulk/util/config/Compat.java

```diff
@@ -105,6 +105,10 @@ private static Optional<String> callToJsonIfAvailable(final DataSource source) {
             jsonStringObject = toJson.invoke(source);
         } catch (final InvocationTargetException ex) {
             final Throwable targetException = ex.getTargetException();
+            if (targetException instanceof UnsupportedOperationException) {
+                // If the plugin's embulk-util-config does not implement toJson, it cannot retrieve a stringified JSON.
```

Comment:
What do you think to add logging message?

Reply:
This part is still an "expected" path, then I'm afraid that logging here could be noisy.

Reply:
Thanks! LGTM👍